src/supremm/datasource/pcp/pcpsummarize.py

```diff
@@ -0,0 +1,365 @@
+        self.archives_processed = 0
+        self.rangechange = RangeChange(config)
+
+    def process(self):
+        """ Main entry point. All archives are processed """
+        success = 0
+        self.archives_processed = 0
+
+        for nodename, nodeidx, archive in self.job.nodearchives():
+            try:
+                self.processarchive(nodename, nodeidx, archive)
+                self.archives_processed += 1
+            except pmapi.pmErr as exc:
+                success -= 1
+                #pylint: disable=not-callable
+                self.adderror("archive", "{0}: pmapi.pmErr: {1}".format(archive, exc.message()))
+
+            except Exception as exc:
+                success -= 1
+                self.adderror("archive", "{0}: Exception: {1}. {2}".format(archive, str(exc), traceback.format_exc()))
+                if self.fail_fast:
+                    raise
+
+        return success == 0
+
+    def complete(self):
+        """ A job is complete if archives exist for all assigned nodes and they have
+            been processed sucessfullly
+        """
+        return self.job.nodecount == self.archives_processed
+
+    def good_enough(self):
+        """ A job is good_enough if archives for 95% of nodes have
+            been processed sucessfullly
+        """
+        return self.archives_processed >= 0.95 * float(self.job.nodecount)
+
+    def get(self):
+        """ Return a dict with the summary information """
+        output = {}
+        timeseries = {}
+
+        je = self.job.get_errors()
+        if len(je) > 0:
+            self.adderror("job", je)
+
+        if self.job.nodecount > 0:
+            for analytic in self.alltimestamps:
+                if analytic.status != "uninitialized":
+                    if analytic.mode == "all":
+                        output[analytic.name] = analytic.results()
+                    if analytic.mode == "timeseries":
+                        timeseries[analytic.name] = analytic.results()
+            for analytic in self.firstlast:
+                if analytic.status != "uninitialized":
+                    output[analytic.name] = analytic.results()
+
+        output['summarization'] = {
+            "version": VERSION,
+            "elapsed": time.time() - self.start,
+            "created": time.time(),
+            "srcdir": self.job.jobdir,
+            "complete": self.complete()}
+
+        output['created'] = datetime.datetime.utcnow()
+
+        output['acct'] = self.job.acct
+        output['acct']['id'] = self.job.job_id
+
+        if len(timeseries) > 0:
+            timeseries['hosts'] = dict((str(idx), name) for name, idx, _ in self.job.nodearchives())
+            timeseries['version'] = TIMESERIES_VERSION
+            output['timeseries'] = timeseries
+
+        for preproc in self.preprocs:
+            result = preproc.results()
+            # Don't overrwrite proc data if ProcPrometheus is enabled
+            if preproc.name == 'procprom':
```

Comment:
Please see other comments about the pre-procs. We should not have prometheus-specfic code in the pcp code.

Reply:
I have a fix for this in https://github.com/ubccr/supremm/pull/259/commits/7ac9f16c393045118491a9a9ccdadc678f1d148a. I removed these lines and successfully tested this change on Rocky 8 Ookami. The previous issue that the code above meant to resolve is now resolved by the new commit.

Reply:
This was also tested with Prometheus data on supremm-dev in addition to the PCP data on Rocky 8 Ookami.